Sporadic test failure in `normalizeURI`

This test failure has occurred a couple of times recently:

```
FAILED TESTS:
  annotator.util.url
    normalizeURI
      ✖ resolves relative URLs against the document URI, if no base URI is provided
        Chrome Headless 107.0.5304.87 (Linux x86_64)
      AssertionError: expected 'http://localhost:9876/books/abc/epub/…' to equal 'http://localhost:9876/books/abc/epub/…'

      + expected - actual

      -http://localhost:9876/books/abc/epub/OPS/xhtml/foo.html
      +http://localhost:9876/books/abc/epub/OPS/xhtml/chapter_001.html?ignoreme#cfi=/foo/foo.html
      
    at Context.<anonymous> (/home/runner/work/client/client/src/annotator/util/test/url-test.js:13:14)
```

From Slack thread: https://hypothes-is.slack.com/archives/C1M8NH76X/p1669119743405089

--- a/src/annotator/integrations/test/vitalsource-test.js
+++ b/src/annotator/integrations/test/vitalsource-test.js
@@ -496,8 +496,14 @@ describe('annotator/integrations/vitalsource', () => {
         history.pushState({}, '', bookURI);
       });
 
-      afterEach(() => {
+      afterEach(async () => {
+        const urlChanged = new Promise(resolve => {
+          window.addEventListener('popstate', () => resolve(), { once: true });
+        });
+
         history.back();
+
+        await urlChanged;
       });
 
       context('when "book_as_single_document" flag is off', () => {

